daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -480,7 +504,8 @@ trait ElementBaseGrammarMixin
     case LengthKind.Implicit => implicitBinaryLengthInBits
```

Comment:
Adding the Asserts does break things.
The first Assert breaks HexBinary, because I believe it is handled as a string
The second Assert breaks the HexBinary calendar as it is a date/time not Numeric.Kind